### PR TITLE
consensus: change `process_past_events` to take message ownership

### PR DESCRIPTION
**Pull request by herr-seppia**

### CODE REVIEW

File: consensus/src/execution_ctx.rs [hunk 1]
```diff
@@ -286,40 +286,30 @@ impl<'a, DB: Database, T: Operations + 'static> ExecutionCtx<'a, DB, T> {
...
+        let collected = phase
             .lock()
             .await
-            .collect(msg.clone(), &self.round_update, committee)
+            .collect(msg, &self.round_update, committee)
             .await;
 
-        match ret {
-            Ok(output) => {
-                trace!("message collected {:#?}", msg);
```
Comment (goshawk-3) [inline consensus/src/execution_ctx.rs:297]: Can we keep this trace or maybe it's redundant?

File: consensus/src/execution_ctx.rs [hunk 2]
```diff
@@ -189,7 +189,7 @@ impl<'a, DB: Database, T: Operations + 'static> ExecutionCtx<'a, DB, T> {
     }
 
     /// Process messages from past
-    async fn process_past_events(&mut self, msg: &Message) -> Option<Message> {
+    async fn process_past_events(&mut self, msg: Message) -> Option<Message> {
```
Comment (goshawk-3) [inline consensus/src/execution_ctx.rs:192]: Good catch!

**Review by goshawk-3 — APPROVED** [review]
LGTM

**Review by fed-franz — APPROVED** [review]
LGTM

File: consensus/src/execution_ctx.rs (hunk 1 above)
Comment (herr-seppia) [reply consensus/src/execution_ctx.rs:297]: I moved 3lines above

**Pull request merged**